File: front_pfe/src/app/model/besoin.ts
import { BesoinFile } from "./besoin-file";
export class Besoin {
    idBesoin: number;
    filBesoin : string;
    date : Date;
    posteBesoin : string;
    descPoste : string;
    service : string;
    motifBesoin : string;
    dateEmb : Date;
    statBesoin : string;
    besoin : BesoinFile;

    constructor(idbesoin: number,filBesoin : string,date : Date,posteBesoin : string,descPoste : string,
        service : string,motifBesoin : string,dateEmb : Date,statBesoin : string,besoin : BesoinFile ) {

            this.idBesoin = idbesoin;
            this.filBesoin = filBesoin;
            this.date = date;
            this.posteBesoin = posteBesoin;
            this.descPoste = descPoste;
            this.service = service;
            this.motifBesoin = motifBesoin;
            this.dateEmb = dateEmb;
            this.statBesoin = statBesoin;
            this.besoin = besoin;
            
    }
}



                

File: back_pfe/src/main/java/com/app/entity/Besoin.java
package com.app.entity;

import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import javax.persistence.*;
import java.util.Date;

@AllArgsConstructor
@Entity
@Table(name= "BESOIN_TBL")
@Getter
@Setter
@ToString
public class Besoin {
    @Id
    @GeneratedValue
    private int idBesoin;
    private String filBesoin;
    private Date date;
    private String posteBesoin;
    private String descPoste;
    private String service;
    private String motifBesoin;
    private Date dateEmb;
    private String statBesoin;
    @ManyToOne
    private FileDB besoin;

    public Besoin() {

    }
}
